### DISCUSSION TITLE
how to connect bootstrap 5

### DISCUSSION THREAD

**Discussion by SchwarzAlis** [vuejs/vitepress#2393] [Q&A]
how to connect bootstrap 5 to vitepress and keep the default theme working

**Comment by zangetsu02** [comment]
any info?

**Comment by SimonEast** [comment]
Well be aware that Bootstrap's styles (particularly the base styling in "reboot") may conflict with some of the default styles in VitePress, so you may get a few weird results that require some troubleshooting. But it's probably somewhat necessary if you desire to use any Bootstrap components within your VuePress site. 

Thankfully, if you use this method, you can import just the parts of Bootstrap you want and also adjust any Bootstrap variables where needed.

The way I would do it is:

1. Install Bootstrap and SASS by running this command in your VitePress project folder:

   ```sh
   npm install bootstrap sass
   ```

3. Create a VitePress theme that extends the default theme by putting this code into `.vitepress/theme/index.js`:

   ```js
   import DefaultTheme from 'vitepress/theme'
   import './custom.scss'
   export default DefaultTheme
   ```

4. Import the Bootstrap SASS items you need by copying and pasting `node_modules/bootstrap/scss/bootstrap.scss` into `.vitepress/theme/custom.scss` and adding `bootstrap/scss/` to the path of each of the imports. Comment out the components that you don't need.

   It will look something like this (but this is only a short snippet of the full code):

   ```js
   // Configuration
   @import "bootstrap/scss/functions";
   @import "bootstrap/scss/variables";
   @import "bootstrap/scss/variables-dark";
   @import "bootstrap/scss/maps";
   @import "bootstrap/scss/mixins";
   @import "bootstrap/scss/utilities";
   
   // Layout & components
   @import "bootstrap/scss/root";
   @import "bootstrap/scss/reboot";
   // @import "bootstrap/scss/type";
   // @import "bootstrap/scss/images";
   // @import "bootstrap/scss/containers";
   // @import "bootstrap/scss/grid";
   // @import "bootstrap/scss/tables";
   // @import "bootstrap/scss/forms";
   // @import "bootstrap/scss/buttons";
   ...
   ```

5. I have not attempted this, but if you also require Bootstrap JS libraries, you can try injecting them into the head of the template. See [this site config explanation](https://vitepress.dev/reference/site-config#head).

   Or you may be able to create a [custom theme entry file](https://vitepress.dev/guide/custom-theme) and specify the components you need, like:

   ```js
   import Dropdown from 'bootstrap/js/dist/dropdown'
   ```

Hope that helps.

**Reply by matevoux** [reply]
check theme using bootstrap like [modern blog](https://github.com/z3by/vuepress-theme-modern-blog)
